src/Avalonia.Controls/Border.cs

```diff
@@ -169,13 +169,36 @@ public BoxShadows BoxShadow
             set => SetValue(BoxShadowProperty, value);
         }
 
+        private Thickness _layoutThickness = default;
+
+        private Thickness LayoutThickness
+        {
+            get
+            {
+                if (_layoutThickness == default)
```

Comment:
I don't see, but does it get invalidated when border thickness or layout scaling is changed?

Reply:
Great point. Passing the Layout scaling itself would require some quite complex additions in order that it correctly gets passed from native. I didn't want to touch any of the interfaces like ILayoutRoot so what I did was cache the scale on first get and then checking has it changed on each render. If you have a better idea how to do this I am open to suggestions.

Also I noticed that this might change depending on two other properties: UseLayoutRounding and BorderThickness so I added the invalidation in OnPropertyChanged.